### PR TITLE
update to make schemes flat

### PR DESCRIPTION
**Pull request by bbakernoaa**
Make the schemes flat to better enable porting of other code to CATChem

### CODE REVIEW

File: src/process/dust/ccpr_scheme_fengsha_mod.F90 [hunk 1]
```diff
@@ -34,36 +34,79 @@ module CCPr_Scheme_Fengsha_Mod
...
+   !! \param DragOpt Drag option
+   !! \param HorizFluxOpt Horizontal flux option
+   !! \param TotalEmission Total Emission Rate [ug/m2/s]
+   !! \param EmissionBin Emission Rate per Species [ug/m2/s]
+   !! \param RC Return code
    !!
    !! \ingroup catchem_dust_process
    !!!>
-   subroutine CCPr_Scheme_Fengsha(MetState, DiagState, DustState, RC)
+   subroutine CCPr_Scheme_Fengsha(nDustSpecies, DSOILTYPE, SSM, RDRAG, TSKIN, USTAR, USTAR_THRESHOLD, GWETTOP, z0, &
```
Comment (zmoon) [inline src/process/dust/ccpr_scheme_fengsha_mod.F90]: I think it would be easier to compare to the call if this were one on each line too.

File: src/process/dust/ccpr_scheme_fengsha_mod.F90 [hunk 2]
```diff
@@ -34,36 +34,79 @@ module CCPr_Scheme_Fengsha_Mod
...
+      CLAYFRAC, SANDFRAC, AIRDEN, FROCEAN, FRLANDICE, FRSNOW, alpha, gamma, &
+      reff, rlower, rupper, MoistOpt, DragOpt, HorizFluxOpt, TotalEmission, EmissionBin, RC)
```
Comment (zmoon) [inline src/process/dust/ccpr_scheme_fengsha_mod.F90]: The opt ones could be `optional` parameters

File: tests/test_dust.f90 [hunk 3]
```diff
@@ -94,7 +94,7 @@ program test_dust
    end if
 
    call print_info(Config, DustState, MetState, title)
-   call assert(DiagState%dust_total_flux > 1000.0_fp, "Test Fengsha Dust Scheme")
+   call assert(DiagState%dust_total_flux > 0.0_fp, "Test Fengsha Dust Scheme")
```
Comment (zmoon) [inline tests/test_dust.f90]: Why this change?

Comment (bbakernoaa) [reply tests/test_dust.f90]: mistake

File: src/process/seasalt/ccpr_seasalt_common_mod.F90 [hunk 4]
```diff
@@ -195,7 +195,7 @@ subroutine weibullDistribution(gweibull, weibullFlag, wm, RC)
       ! Initialize
       errMsg = ''
       thisLoc = ' -> at weibullDistribution (in util/metutils_mod.F90)'
-      RC = CC_SUCCESS
+      RC = -1
```
Comment (zmoon) [inline src/process/seasalt/ccpr_seasalt_common_mod.F90]:
```suggestion
      RC = 0
```

File: src/process/seasalt/ccpr_seasalt_common_mod.F90 [hunk 5]
```diff
@@ -212,11 +212,7 @@ subroutine weibullDistribution(gweibull, weibullFlag, wm, RC)
          endif
       endif
 
-      if (RC /= CC_SUCCESS) then
-         errMsg = 'Error Calculating Weibull Distribution'
-         call CC_Error(errMsg, RC, thisLoc)
-         return
-      endif
+      RC = 0
```
Comment (zmoon) [inline src/process/seasalt/ccpr_seasalt_common_mod.F90]:
```suggestion
```

File: src/process/seasalt/ccpr_scheme_gong97_mod.F90 [hunk 6]
```diff
@@ -172,21 +204,21 @@ subroutine CCPr_Scheme_Gong97(MetState, DiagState, SeaSaltState, RC)
...
             MassEmissions = ZERO
             NumberEmissions = ZERO
 
          enddo ! nbins
 
-         SeaSaltState%TotalEmission = sum(SeaSaltState%EmissionPerSpecies)
-         DiagState%sea_salt_total_flux = SeaSaltState%TotalEmission
-         SeaSaltState%TotalNumberEmission = sum(SeaSaltState%NumberEmissionBin)
+         TotalEmission = sum(EmissionBin)
+         ! DiagState%sea_salt_total_flux = TotalEmission
```
Comment (zmoon) [inline src/process/seasalt/ccpr_scheme_gong97_mod.F90]:
```suggestion
```

File: src/process/seasalt/ccpr_scheme_gong97_mod.F90 [hunk 7]
```diff
@@ -127,29 +151,37 @@ subroutine CCPr_Scheme_Gong97(MetState, DiagState, SeaSaltState, RC)
          exppow   = 1.19_fp
          wpow     = 3.41_fp
 
-         ! Weibull Distribution following Fan and Toon 2011 if SeaSaltState%WeibullFlag
+         ! Weibull Distribution following Fan and Toon 2011 if WeibullFlag
          !----------------------------------------------------------------------------
-         call weibullDistribution(gweibull, SeaSaltState%WeibullFlag, w10m, RC)
+         call weibullDistribution(gweibull, WeibullFlag, w10m, RC)
+         if (RC /= 0) then
+            RC = -1
```
Comment (zmoon) [inline src/process/seasalt/ccpr_scheme_gong97_mod.F90:158]: print error message about Weibull before returning?

File: src/process/dust/ccpr_dust_common_mod.F90 [hunk 8]
```diff
@@ -9,7 +9,7 @@
 !!!>
 module CCPr_Dust_Common_Mod
    use precision_mod, only: fp, ZERO
-   use Error_Mod
+   ! use Error_Mod
```
Comment (zmoon) [inline src/process/dust/ccpr_dust_common_mod.F90]:
```suggestion
```

**Comment by bbakernoaa** [comment]
@zmoon this one too

File: src/process/dust/CCPr_Dust_Mod.F90 [hunk 9]
```diff
@@ -237,13 +237,53 @@ SUBROUTINE CCPr_Dust_Run( MetState, DiagState, DustState, ChemState, RC )
...
+               MetState%GWETTOP,                &
+               MetState%z0,                     &
+               MetState%CLAYFRAC,               &
+               MetSTate%SANDFRAC,               &
+               MetState%AIRDEN(1),              &
+               MetState%FROCEAN,                 &
+               MetState%FRLANDIC,               &
+               MetState%FRSNO,                  &
+               DustState%AlphaScaleFactor,      &
+               DustState%BetaScaleFactor,       &
```
Comment (zmoon) [inline src/process/dust/CCPr_Dust_Mod.F90:256]: its gamma in the subroutine, should it be gamma here instead of beta?

File: src/process/dust/ccpr_scheme_fengsha_mod.F90 [hunk 10]
```diff
@@ -34,36 +34,102 @@ module CCPr_Scheme_Fengsha_Mod
...
+      real(fp), intent(in) :: GWETTOP         !< Top soil wetness [m3/m3]
+      real(fp), intent(in) :: z0              !< Surface roughness [m]
+
+      real(fp), dimension(:), intent(in) :: reff            !< Effective Radius
+      real(fp), dimension(:), intent(in) :: rlower          !< Lower Radius
+      real(fp), dimension(:), intent(in) :: rupper         !< Upper Radius
+
+      integer,  intent(in) :: MoistOpt        !< Option for moisture
+      integer,  intent(in) :: DragOpt         !< Option for drag
+      integer,  intent(in) :: HorizFluxOpt    !< Option for horizontal flux
```
Comment (zmoon) [inline src/process/dust/ccpr_scheme_fengsha_mod.F90]: Still think you could make these `optional`

**Review by zmoon — COMMENTED** [review]
I think it's looking fine. Perhaps one could be left in the passing state types format though, so we have an example of that?

File: src/process/dust/CCPr_Dust_Mod.F90 (hunk 9 above)
Comment (bbakernoaa) [reply src/process/dust/CCPr_Dust_Mod.F90:256]: 🤷

File: src/process/dust/ccpr_scheme_fengsha_mod.F90 (hunk 10 above)
Comment (bbakernoaa) [reply src/process/dust/ccpr_scheme_fengsha_mod.F90]: made these optional

File: src/process/dust/ccpr_scheme_fengsha_mod.F90 [hunk 11]
```diff
@@ -163,6 +163,10 @@ subroutine CCPr_Scheme_Fengsha(nDustSpecies,    &
 
       nbins = size(reff)
 
+      if (present(MoistOpt) .eqv. .false.) MoistOpt = 1
+      if (present(DragOpt) .eqv. .false.) DragOpt = 1
+      if (present(HorizFluxOpt) .eqv. .false.) HorizFluxOpt = 1
```
Comment (zmoon) [inline src/process/dust/ccpr_scheme_fengsha_mod.F90]: Unfortunately, you really need to have a separate variable for this to work properly in both cases.

```
integer, intent(in), optional :: MoistOpt  ! optional input

integer :: MoistOpt_  ! local variable to actually use

MoistOpt_ = 1  ! default
if (present(MoistOpt)) MoistOpt_ = MoistOpt  ! override default if provided

! ...
! Use `MoistOpt_` in the code
```

File: src/process/dust/ccpr_scheme_fengsha_mod.F90 [hunk 12]
```diff
@@ -34,36 +34,102 @@ module CCPr_Scheme_Fengsha_Mod
...
+      FRLANDICE,        &
+      FRSNOW,           &
+      alpha,            &
+      gamma,            &
+      reff,             &
+      rlower,           &
+      rupper,           &
+      MoistOpt,         &
+      DragOpt,          &
+      HorizFluxOpt,     &
```
Comment (zmoon) [inline src/process/dust/ccpr_scheme_fengsha_mod.F90]: Convention would be, I believe, to move this to the end of list, and pass by name (`DragOpt=..., ...`) in the call.

File: src/process/seasalt/ccpr_scheme_gong97_mod.F90 (hunk 7 above)
Comment (zmoon) [reply src/process/seasalt/ccpr_scheme_gong97_mod.F90:158]: otherwise it would be hard to track down the issue

File: src/process/dust/CCPr_Dust_Mod.F90 (hunk 9 above)
Comment (zmoon) [reply src/process/dust/CCPr_Dust_Mod.F90:256]: 🤷‍♂️

File: src/process/seasalt/ccpr_scheme_gong97_mod.F90 (hunk 7 above)
Comment (bbakernoaa) [reply src/process/seasalt/ccpr_scheme_gong97_mod.F90:158]: I am fine doing so but we don't import error_mod at the scheme level.  We could write something simple to write out an error message

Comment (zmoon) [reply src/process/seasalt/ccpr_scheme_gong97_mod.F90:158]: It could just be something simple like `print *, "ahhh Weibull"`

**Comment by bbakernoaa** [comment]
@zmoon This one too

**Comment by bbakernoaa** [comment]
@zmoon bump

**Comment by zmoon** [comment]
@bbakernoaa what did you think about this comment?

> Perhaps one [process] could be left in the passing state types format though, so we have an example of that?

Otherwise I think my comments have been addressed.

**Comment by bbakernoaa** [comment]
> @bbakernoaa what did you think about this comment?
> 
> > Perhaps one [process] could be left in the passing state types format though, so we have an example of that?
> 
> Otherwise I think my comments have been addressed.

I say we not worry about it and just make it so this is the new format for now

**Review by zmoon — APPROVED** [review]
